walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -27,10 +27,33 @@
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        List<TimeRange> mandatoryAvailableRanges = new ArrayList<>();
+        List<TimeRange> optionalAttendeesConsideredRanges = new ArrayList<>();
+
+        /* try to remove the timeRanges optional attendees can't make it to. If there's nothing left, ignore the optional attendees*/
+        mandatoryAvailableRanges = availableRanges(events, request, false);
```

Comment:
If you perform some of the checks earlier, you can avoid doing work that you don't need to:
```
List<TimeRange> optionalAttendeesConsideredRanges = availableRanges(events, request, true);
if (request.getAttendees().isEmpty()) {
  return optionalAttendeesConsideredRanges;
}

List<TimeRange> mandatoryAvailableRanges = availableRanges(events, request, false);
...
```

Reply:
Thanks